omnipaxos_core/src/sequence_paxos/mod.rs

```diff
@@ -62,28 +62,47 @@ where
         let majority = num_nodes / 2 + 1;
         let max_peer_pid = peers.iter().max().unwrap();
         let max_pid = *std::cmp::max(max_peer_pid, &pid) as usize;
-        let (state, leader, lds) = match &config.skip_prepare_use_leader {
-            Some(l) => {
-                let (role, lds) = if l.pid == pid {
-                    // we are leader in new config
-                    let mut v = vec![None; max_pid];
-                    for idx in peers.iter().map(|pid| *pid as usize - 1) {
-                        // this works as a promise
-                        v[idx] = Some(0);
-                    }
-                    (Role::Leader, Some(v))
-                } else {
-                    (Role::Follower, None)
-                };
-                let state = (role, Phase::Accept);
-                (state, *l, lds)
+        let mut outgoing = Vec::with_capacity(BUFFER_SIZE);
+        let mut state = (Role::Follower, Phase::None);
+        let mut leader = Ballot::default();
+        let mut lds = None;
```

Comment:
Do these really need to be mutable? Can't we just initialize them as before i.e. `let (state, leader, lds) = match { ... }`